include/c_utilities/types/string_array.h

```diff
@@ -80,7 +80,7 @@ utilities_get_pre_initialized_string_array(size_t size);
  */
 C_UTILITIES_PUBLIC
 utilities_ret_t
```

Comment:
This function should also have a [`C_UTILITIES_WARN_UNUSED`](https://github.com/ros2/c_utilities/blob/5711a36d29ba6245d4231ac75f8c5f7e1862c3ba/include/c_utilities/macros.h#L23-L27).